test/com/sun/jna/ArgumentsMarshalNullableTest.java

```diff
@@ -32,31 +32,31 @@
  * and result in deterministic behaviour and not in a JVM crash.
  */
 public class ArgumentsMarshalNullableTest extends TestCase {
-    public static class Int32Integer extends IntegerType {
+    public static class Int16Integer extends IntegerType {
```

Comment:
Yes this was intentional. Some background: I added the test when I implemented nullable IntegerTypes in the direct invoke path in 1a2a94f. I used the mapped function because they used the fixed size int32_t. Reduzing the size should expose more systems, that show problematic behavior. If sizeof(targettype) >= sizeof(ffi_arg) we will see no problem, so reducing the size of the argument makes it more likely to hit the problematic path. I see no benefit keeping both variants.